scanpy/tools/_leiden.py

```diff
@@ -134,22 +137,38 @@ def leiden(
             adjacency=adjacency,
         )
     # convert it to igraph
+    if use_igraph and directed:
+        raise ValueError(
+            "Cannot use igraph's leiden implementaiton with a directed graph."
+        )
     g = _utils.get_igraph_from_adjacency(adjacency, directed=directed)
     # flip to the default partition type if not overriden by the user
-    if partition_type is None:
+    if partition_type is None and not use_igraph:
         partition_type = leidenalg.RBConfigurationVertexPartition
+    elif use_igraph and partition_type is not None:
+        raise ValueError("Do not pass in partition_type argument when using igraph.")
     # Prepare find_partition arguments as a dictionary,
     # appending to whatever the user provided. It needs to be this way
     # as this allows for the accounting of a None resolution
     # (in the case of a partition variant that doesn't take it on input)
     if use_weights:
-        partition_kwargs["weights"] = np.array(g.es["weight"]).astype(np.float64)
-    partition_kwargs["n_iterations"] = n_iterations
-    partition_kwargs["seed"] = random_state
+        clustering_args["weights"] = (
+            "weight" if use_igraph else np.array(g.es["weight"]).astype(np.float64)
+        )
```

Comment:
I think both of these should be added to `uns` as outputs, `weights` in the `params` part of the dict and `use_igraph` at the top levels

```python
adata['leiden']
# { 'params': { ... 'use_weights': True }, 'use_igraph': False }